src/adapter/src/client.rs

```diff
@@ -532,9 +545,19 @@ impl SessionClient {
                     results.push(SimpleResult::err(
                         "executing statements of this type is unsupported via this API",
                     ));
+                    break;
```

Comment:
I'm not convinced the `breaks` do the correct thing with transactions. There's 3 places they occur at the end here, but none of them fail the transaction, and so they'll attempt to commit it.

I think this would be safer if the "execute exactly one query" part were it's own function, and the caller could inspect for error and do something. Is that refactor reasonable?

Reply:
These were unintentional, however, I've gone ahead and refactored this to unify error handling in the loop.